packages/web/nexus/subscription.ts

```diff
@@ -0,0 +1,322 @@
+import { MembershipSubscriptionPeriod, InputJsonValue, PrismaClient } from '@journaly/j-db-client'
+import {
+  arg,
+  booleanArg,
+  extendType,
+  objectType,
+  stringArg,
+} from '@nexus/schema'
+import stripe from './utils/stripe'
+
+const MembershipSubscription = objectType({
+  name: 'MembershipSubscription',
+  definition(t) {
+    t.model.id()
+    t.model.period()
+    t.model.userId()
+    t.model.expiresAt()
+    t.model.cancelAtPeriodEnd()
+    t.model.lastFourCardNumbers()
+    t.model.cardType()
+  }
+})
+
+const getSubscriptionPriceId = (subType: MembershipSubscriptionPeriod) => {
+  switch(subType) {
+    case MembershipSubscriptionPeriod.MONTHLY:
+      return 'price_1ISRgvB8OEjVdGPaQr7ZANW8'
+    case MembershipSubscriptionPeriod.QUARTERLY:
+      return 'price_1ISRgvB8OEjVdGPaeOx4m255'
+    case MembershipSubscriptionPeriod.ANNUALY:
+      return 'price_1ISRgvB8OEjVdGPam1PTr6hE'
+  }
+}
+
+const setPaymentMethod = async (
+  userId: number,
+  db: PrismaClient,
+  customerId: string,
+  paymentMethodId: string,
+  lastFourCardNumbers: string = '1234',
+  cardType: string = 'Visa'
+) => {
+  const customer = await stripe.customers.retrieve(customerId)
+  
+  if (!customer) throw new Error("User has stripeCustomerId but unable to find customer in Stripe")
+
+  await stripe.paymentMethods.attach(paymentMethodId, {
+    customer: customer.id,
+  })
+
+  // Update customer's default method in case they've entered a new card
```

Comment:
think this comment doesn't apply anymore

Reply:
Gone